src/js/button.js

```diff
@@ -7,13 +12,13 @@
  * @class
  * @constructor
  */
-vjs.Button = vjs.Component.extend({
+var Button = Component.extend({
   /**
    * @constructor
    * @inheritDoc
    */
   init: function(player, options){
```

Comment:
I really like the Idea of moving to ES6 classes but I am worried about the amount of work it will take to migrate from 4.x.x to 5.x.x.  

Once this branch is finally merged I will migrate the player we are using at MailOnline to see how much effort it takes. 
Are you already writing a migration guide for 5.0.0 if not I can start writing one while I migrate our player.